src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -1272,6 +1272,12 @@ def log_message(self, format, *args):  # pylint: disable=redefined-builtin,unuse
         results['no_browser'] = True
         return
 
+    # Emit a warning to inform that a browser is opened.
+    # Only show the path part of the URL and hide the query string.
+    logger.warning("The default web browser has been opened at %s. Please continue the login in the web browser.\n"
+                   "If no web browser is available or if the web browser fails to open, use device code flow "
```

Comment:
If the web browser does fail to open, will CLI run into this line or hang over line 1269 `succ = open_page_in_browser(url)` ?

Reply:
If the web browser fails to open, `open_page_in_browser` should return `False`. This warning is for the scenario where `open_page_in_browser` returned `True`, but the browser is not brought to the foreground for some reason. (This mainly happens on some Linux distributions or MacOS.)